### PR TITLE
Post libtxt/post iOS 11 fidelity fine tuning

### PR DESCRIPTION
**Pull request by xster**
Matched with screenshots of native apps

### CODE REVIEW

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm

**Comment by xster** [comment]
cc @tvolkert @jason-simmons , I tried to repro the tests on Linux. 

Different visible here:
download
 
It seems like there's a minute difference in font rendering here to cause an error from the golden.

**Comment by tvolkert** [comment]
@xster it looks like we should enable the `--skia-deterministic-rendering` when running via `flutter test`

**Comment by tvolkert** [comment]
I've ruled out <span>#</span>1 and <span>#</span>2 from above (the pixel values are indeed different, and `flutter_tester` _is_ respecting the `--skia-deterministic-rendering` flag).  But there are some additional possibilities that occurred to me:

4. shadows will always be slightly different across platforms, and once https://github.com/flutter/flutter/pull/17363 lands, the tests will pass.
4. The host embedder (`flutter_tester`) doesn't respect `--enable-software-rendering`, thus making the `--skia-deterministic-rendering` flag a no-op.

**Comment by tvolkert** [comment]
Looking at the images, it doesn't appear as if shadows are even in play here, so <span>#</span>4 can be safely ruled out...

**Comment by tvolkert** [comment]
Zoomed up on the upper right corner of one of those boxes, here's what the diff looks like:

nav_bar_test large_title 1

That's a noticeable rendering difference, so although it's unclear where the bug is, it looks like this did catch a potentially real issue...

**Comment by xster** [comment]
I wonder if we could make the issue clearer if we used another proper font that's available on both platforms.

**Comment by tvolkert** [comment]
@chinmaygarde says that the screenshotter always uses software rendering, so I'm fresh out of ideas beyond <span>#</span>3 (Skia isn't deterministic even when `--skia-deterministic-rendering` is specified)

File: packages/flutter/test/cupertino/nav_bar_test.dart [hunk 1]
```diff
@@ -481,6 +481,68 @@ void main() {
...
+                  child: const Center(),
+                ),
+              );
+            },
+          );
+        },
+      ),
+    );
+
+    await expectLater(
```
Comment (tvolkert) [inline packages/flutter/test/cupertino/nav_bar_test.dart]: Was discussing this with @Hixie.  For now at least, we'll skip the golden file tests on non-Linux platforms (and require that developers update their goldens while running on Linux).

So tldr: add a `skip: !Platform.isLinux` to these new `testWidgets()` methods

Comment (xster) [reply packages/flutter/test/cupertino/nav_bar_test.dart]: cool, thanks for following up. I'll update the goldens

**Review by tvolkert — APPROVED** [review]

**Pull request merged**